### ISSUE TITLE
Ability to interrupt a moveto on actors

### ISSUE THREAD

**Issue by blurymind** [chrismaltby/gb-studio#1019]
**Is your feature request related to a problem? Please describe.**
Lets say I have an AI with a patrooling behavior

The AI is using a switch to jump between states. The patrooling state is using a chain of movetos to define the route
Lets call that state == 1

Now lets say we have the player shoot at the ai, while the ai is patrooling and instead of hiding the enemy actor, we want to put the enemy actor into a temporary stun state.
So when the player bullet hits the enemy actor, we set the ai state to == 2 (stunned). When set to stunned, we expect the enemy to stop moving instantly upon being hit by the bullet. Since the state is now not one, we expect the movetos to instantly stop.

So what appears to happen however is that the ai switches to another state (2), but the moveto continues to execute - after being hit by a bullet the enemy continues to walk, until executing all the remaining movetos - it walks for a while on its route and then finally stops.

**Describe the solution you'd like**
This looks like a bug to me. If the wrapping if statement stops being true, the movetos inside it should instantly be interupted, but they arent.
If there is a plausible technical or design reason for that, then I would like to request an action, which clears the movetos and instantly stops the ais movetos from executing.

**Describe alternatives you've considered**
I "wait" action, when reached should also be capable of stopping/interrupting the movetos- but it seems completely helpless too. 

**Additional context**
Please make a more capable example of enemy ai as part of the project. Gb-studio appears to have a serious flaw preventing it from being able to do a capable state machine using the move actions predictably

**Comment by maxoakland** [comment]
Maybe there could be a “Stop Move” and “Stop Wait” event like the one to stop an actor’s On Update script

**Comment by blurymind** [comment]
the problem with the action to stop an actor’s On Update script is that we dont have one to start it again.

I tried to use that and it did stop the movetos, but I couldnt get the ai to get out of the stun status. We do not have an action to start the on update script after it was stopped

**Comment by maxoakland** [comment]
> the problem with the action to stop an actor’s On Update script is that we dont have one to start it again.
> 
> I tried to use that and it did stop the movetos, but I couldnt get the ai to get out of the stun status. We do not have an action to start the on update script after it was stopped
> 
> it could potentially be also used for pausing the game?

I’m not saying you should use Stop On Update, I’m saying a similar script could be a way to implement this request

**Comment by blurymind** [comment]
yes of course, I like the idea of a "stop move to actions" script, just not sure how to implement it

**Comment by maxoakland** [comment]
> yes of course, I like the idea of a "stop move to actions" script, just not sure how to implement it

If you’re interested in trying it yourself the discord has a lot of people who are knowledgeable and helpful in the plugins-help channel

**Comment by blurymind** [comment]
no I dont think I can do it myself, I have too little knowledge about the engine api

**Comment by blurymind** [comment]
@um3k thank you for getting this in 🎉 
I am gonna have some fun with it over the weekend :)

**Comment by um3k** [comment]
GBVM is the game engine itself, gb-studio is the editor. It contains a copy of the gbvm code, but all the engine changes need to happen in the gbvm repo before being migrated to the editor. The good news is that even if the event doesn't make it into the next version of gb-studio, the GBVM function will be there, so you'll be able to use it in a GBVM script.

**Comment by blurymind** [comment]
is there example code of how to write a gbvm script and use it in gb-studio?
